packages/cli/src/cmds/stats/stats.ts

```diff
@@ -1,335 +1,51 @@
 import yargs from 'yargs';
-import { chdir } from 'process';
-const fs = require('fs');
-import UI from '../userInteraction';
-import Telemetry from '../../telemetry';
-import { listAppMapFiles, verbose } from '../../utils';
-import { Event, buildAppMap } from '@appland/models';
-
-import { AppMapSize, AppMapSizeTable, SortedAppMapSize } from './types/appMapSize';
-import { FunctionExecutionTime, SlowestExecutionTime } from './types/functionExecutionTime';
-import chalk from 'chalk';
-import { relative } from 'path';
+import { statsForDirectory } from './directory/statsForDirectory';
+import { EventInfo, statsForMap } from './directory/statsForMap';
+import { SortedAppMapSize } from './types/appMapSize';
+import { SlowestExecutionTime } from './types/functionExecutionTime';
 
 export const command = 'stats [directory]';
-export const describe = 'Show some statistics about events in scenarios read from AppMap files';
+export const describe =
+  'Show statistics about events from an AppMap or from all AppMaps in a directory';
 const LIMIT_DEFAULT = 10;
-const MINIMUM_APPMAP_SIZE = (1024 * 1024) / 1;
 
 export const builder = (args: yargs.Argv) => {
   args.option('directory', {
-    describe: 'Working directory for the command.',
+    describe: 'Directory to analyze',
     type: 'string',
     alias: 'd',
   });
 
   args.option('json', {
-    describe: 'Format results as JSON.',
+    describe: 'Format results as JSON',
     type: 'boolean',
     alias: 'j',
   });
 
   args.option('limit', {
-    describe: 'limit the number of methods displayed (default ' + LIMIT_DEFAULT + ').',
+    describe: 'Number of methods to display',
     type: 'number',
     alias: 'l',
     default: LIMIT_DEFAULT,
   });
 
+  args.option('appmap-file', {
+    describe: 'AppMap to analyze',
+    type: 'string',
+    alias: 'f',
+  });
```

Comment:
This option is often paired with `--appmap-dir` - one or the other being utilized, and `--appmap-dir` defaulting to the `appmap_dir` found in _appmap.yml_.